### PR TITLE
[Telemetry] Add timing information for the start up sequence

### PR DESCRIPTION
**Pull request by iainx**
Adds timing information to the start up metadata, and copies the start up metadata into the time to code metadata as well, if necessary

Fixes VSTS #652027

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs [hunk 1]
```diff
@@ -160,9 +171,14 @@ int Run (MonoDevelopOptions options)
 			SynchronizationContext.SetSynchronizationContext (DispatchService.SynchronizationContext);
 			Runtime.MainSynchronizationContext = SynchronizationContext.Current;
 
+			sectionTimings["DispatchInitialization"] = startupTimer.ElapsedMilliseconds;
+			startupTimer.Restart ();
+
 			// Initialize Roslyn's synchronization context
 			RoslynServices.RoslynService.Initialize ();
 
+			sectionTimings["RoslynInitialization"] = startupTimer.ElapsedMilliseconds;
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs]: Restart the timer here.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs [hunk 2]
```diff
@@ -736,6 +795,9 @@ public static int Main (string[] args, IdeCustomizer customizer = null)
 				if (!Platform.IsMac && !Platform.IsWindows)
 					exename = exename.ToLower ();
 				Runtime.SetProcessName (exename);
+
+				sectionTimings ["mainInitialization"] = startupTimer.ElapsedMilliseconds;
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/IdeStartup.cs]: restart the stopwatch here?

**Review by Therzok — APPROVED** [review]
LGTM!

**Comment by slluis** [comment]
@monojenkins backport release-7.6

**Comment by slluis** [comment]
@monojenkins backport release-7.7

**Review by slluis — APPROVED** [review]

**Pull request merged**